gobblin-core/src/main/java/gobblin/source/DatePartitionedAvroFileSource.java

```diff
@@ -43,78 +48,98 @@
- * folders following the pattern /my/data/daily/[year]/[month]/[day] are present. It will iterate through all the data
- * under these folders starting from the date specified by {@link #DATE_PARTITIONED_SOURCE_MIN_WATERMARK_VALUE} until
- * either {@link #DATE_PARTITIONED_SOURCE_MAX_FILES_PER_JOB} files have been processed, or until there is no more data
- * to process. For example, if {@link #DATE_PARTITIONED_SOURCE_MIN_WATERMARK_VALUE} is set to 2015/01/01, then the job
- * will read from the folder /my/data/daily/2015/01/01/, /my/data/daily/2015/01/02/, /my/data/2015/01/03/ etc.
- *
+ * The class will iterate through all the data folders given by the base directory
+ * {@link ConfigurationKeys#SOURCE_FILEBASED_DATA_DIRECTORY} and the partitioning type 
+ * {@link #DATE_PARTITIONED_SOURCE_PARTITION_PATTERN} or {@link #DATE_PARTITIONED_SOURCE_PARTITION_GRANULARITY}
+ * 
  * <p>
- *
+ * For example, if the base directory is set to /my/data/ and daily partitioning is used, then it is assumed that
+ * /my/data/daily/[year]/[month]/[day] is present. It will iterate through all the data under these folders starting 
+ * from the date specified by {@link #DATE_PARTITIONED_SOURCE_MIN_WATERMARK_VALUE} until either 
+ * {@link #DATE_PARTITIONED_SOURCE_MAX_FILES_PER_JOB} files have been processed, or until there is no more data 
+ * to process. For example, if {@link #DATE_PARTITIONED_SOURCE_MIN_WATERMARK_VALUE} is set to 2015/01/01, then the job
+ * will read from the folder /my/data/daily/2015/01/02/, /my/data/daily/2015/01/03/, /my/data/2015/01/04/ etc.
+ * </p>
+ * 
+ * </p>
+ * 
  * The class will only process data in Avro format.
  */
-public class DatePartitionedDailyAvroSource extends FileBasedSource<Schema, GenericRecord> {
-
+public class DatePartitionedAvroFileSource extends FileBasedSource<Schema, GenericRecord> {
+  
   // Configuration parameters
-  private static final String DATE_PARTITIONED_SOURCE_PREFIX = "date.partitioned.source.";
-
+  private static final String DATE_PARTITIONED_SOURCE_PREFIX = "date.partitioned.source";
+  
+  public static final String DATE_PARTITIONED_SOURCE_PARTITION_PREFIX =
+      DATE_PARTITIONED_SOURCE_PREFIX + ".partition.prefix";
+  public static final String DATE_PARTITIONED_SOURCE_PARTITION_SUFFIX =
+      DATE_PARTITIONED_SOURCE_PREFIX + ".partition.suffix";
+  public static final String DATE_PARTITIONED_SOURCE_PARTITION_PATTERN =
+      DATE_PARTITIONED_SOURCE_PREFIX + ".partition.pattern";
+  public static final String DATE_PARTITIONED_SOURCE_PARTITION_TIMEZONE =
+      DATE_PARTITIONED_SOURCE_PREFIX + ".partition.timezone";
+  public static final String DEFAULT_DATE_PARTITIONED_SOURCE_PARTITION_TIMEZONE = ConfigurationKeys.PST_TIMEZONE_NAME;
+  public static final String DATE_PARTITIONED_SOURCE_PARTITION_GRANULARITY =
+      DATE_PARTITIONED_SOURCE_PREFIX + ".partition.granularity";
+  public static final DatePartitionType DEFAULT_DATE_PARTITIONED_SOURCE_PARTITION_GRANULARITY =
```

Comment:
Can you make all the `static final` constants `private`? If you want access them from unit tests you can make them package protected and add a `@VisibleForTesting` annotation.